examples/machine_translation/train.py

```diff
@@ -0,0 +1,156 @@
+        trg_char = batch[1][0].transpose(0, 1).to(device)
+        trg_word = batch[1][1].transpose(0, 1).to(device)
+        optimizer.zero_grad()
+
+        output = model(src, trg_char)
+        # Convert prediction to words
+        true = trg_word.transpose(0, 1).unsqueeze(1)
+        true = [[[trg_vocab.itos[word_idx] for word_idx in comb] for comb in sent] for sent in true]
+        pred = output.transpose(0, 1).argmax(-1)
+        pred = [[trg_vocab.itos[word_idx] for word_idx in sent] for sent in pred]
+
+        output = output[1:].reshape(-1, output.shape[-1])
+        trg = trg_word[1:].reshape(-1)
+        loss = criterion(output, trg)
+
+        loss.backward()
+        torch.nn.utils.clip_grad_norm_(model.parameters(), clip)
+        optimizer.step()
+
+        epoch_loss += loss.item()
+        bleu += bleu_score(pred, true)
+
+    return (epoch_loss / len(iterator)), (bleu / len(iterator))
+
+
+def evaluate(model: nn.Module, iterator: DataLoader, criterion: nn.Module, trg_vocab: Vocab, device: torch.device):
+    model.eval()
+
+    epoch_loss = 0
+    bleu = 0.0
+    with torch.no_grad():
+        for _, batch in tqdm(enumerate(iterator), total=len(iterator)):
+            # Need to convert to [seq_len x batch x ...] size
+            src = batch[0].transpose(0, 1).to(device)
+            trg_char = batch[1][0].transpose(0, 1).to(device)
+            trg_word = batch[1][1].transpose(0, 1).to(device)
+
+            output = model(src, trg_char, 0)  # turn off teacher forcing
+            # Convert prediction to words
+            true = trg_word.transpose(0, 1).unsqueeze(1)
+            true = [[[trg_vocab.itos[word_idx] for word_idx in comb] for comb in sent] for sent in true]
+            pred = output.transpose(0, 1).argmax(-1)
+            pred = [[trg_vocab.itos[word_idx] for word_idx in sent] for sent in pred]
+
+            output = output[1:].reshape(-1, output.shape[-1])
+            trg = trg_word[1:].reshape(-1)
+
+            loss = criterion(output, trg)
+
+            epoch_loss += loss.item()
+            bleu += bleu_score(pred, true)
+    return (epoch_loss / len(iterator)), (bleu / len(iterator))
+
+
+if __name__ == "__main__":
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    batch_size = 128
+    train_dataset, val_dataset, test_dataset = get_dataset()
+    train_iterator = DataLoader(train_dataset, batch_size=batch_size, collate_fn=collate_fn)
+    valid_iterator = DataLoader(val_dataset, batch_size=batch_size, collate_fn=collate_fn)
+    test_iterator = DataLoader(test_dataset, batch_size=batch_size, collate_fn=collate_fn)
+    input_dim = len(train_dataset.vocab[0])
+    char_output_dim = len(train_dataset.vocab[1])
+    word_output_dim = len(train_dataset.vocab[2])
+
+    # enc_emb_dim = 256
+    # dec_emb_dim = 256
+    # enc_hid_dim = 512
+    # dec_hid_dim = 512
+    # attn_dim = 64
+    # enc_dropout = 0.5
+    # dec_dropout = 0.5
+
+    enc_emb_dim = 300
```

Comment:
We use `parser` to set up those hyperparameters, like [this](https://github.com/pytorch/text/blob/441edff8743cc03d2d8676029f5bea3c0ed76486/examples/BERT/mlm_task.py#L219)

Reply:
added argparse